diff --git a/wangle/concurrent/Async.h b/wangle/concurrent/Async.h
index 07c18e5..5d9add7 100644
--- a/wangle/concurrent/Async.h
+++ b/wangle/concurrent/Async.h
@@ -15,7 +15,8 @@
 namespace wangle {
 
 template <class F>
-auto async(F&& fn) {
+auto async(F&& fn)
+    -> folly::Future<typename folly::isFuture<decltype(fn())>::Inner> {
   return folly::via<F>(getCPUExecutor().get(), std::forward<F>(fn));
 }
 
